compiler/resolution/initializerResolution.cpp

```diff
@@ -333,6 +333,10 @@ void resolveNewInitializer(CallExpr* newExpr, Type* manager) {
   else if (manager == dtUnmanagedNilable)
     manager = dtUnmanaged;
 
+  if (manager == dtBorrowed) {
+    USR_FATAL(newExpr, "creating a 'new borrowed' type is a not allowed");
```

Comment:
I think this should be a deprecation with a test in test/depracated. Let's fix that in a follow-up.